### PR TITLE
Add smartapp event

### PR DESCRIPTION
**Pull request by Opti213**
# Description

Add Smartapp system event

# Checklist

- [x] Linters and tests have passed
- [ ] I've written a changelog for PR change
- [ ] I'll make a release when PR is approved

### CODE REVIEW

File: botx/models/system_events/smartapp.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        data: Payload.
+        opts: Request options.
+        smartapp_api_version: Protocol version.
+    """
+
+    smartapp_id: UUID
+    data: Dict[str, Any]  # noqa: WPS110
+    smartapp_api_version: int
+    opts: Dict[str, Any] = field(default_factory=dict)
+    ref: Optional[UUID] = None
```
Comment (Nimond) [inline botx/models/system_events/smartapp.py:38]: ref is not optional in incoming smartapp_event

File: botx/models/system_events/smartapp.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        smartapp_id: also personnel chat_id.
+        data: Payload.
+        opts: Request options.
+        smartapp_api_version: Protocol version.
+    """
+
+    smartapp_id: UUID
+    data: Dict[str, Any]  # noqa: WPS110
+    smartapp_api_version: int
+    opts: Dict[str, Any] = field(default_factory=dict)
```
Comment (Nimond) [inline botx/models/system_events/smartapp.py:37]: Maybe remove default value?

File: botx/models/system_events/smartapp.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    command_type: Literal[BotAPICommandTypes.SYSTEM]
+    data: BotAPISmartAppData
+    metadata: Dict[str, Any]
+
+
+class BotAPISmartappEvent(BotAPIBaseCommand):
+    payload: BotAPISmartappPayload = Field(..., alias="command")
+    sender: BotAPIChatContext = Field(..., alias="from")
+
+    source_sync_id: Optional[UUID]
```
Comment (Nimond) [inline botx/models/system_events/smartapp.py:61]: Don't need it in domain, so we can not validate it

File: botx/models/system_events/smartapp.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    entities: List[BotAPIEntity]
```
Comment (Nimond) [inline botx/models/system_events/smartapp.py:62]: Don't need it in domain, so we can not validate it

File: tests/system_events/test_smartapp.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    async with lifespan_wrapper(built_bot) as bot:
+        bot.async_execute_raw_bot_command(payload)
+
+    # - Assert -
+    assert smartapp == SmartAppEvent(
+        bot_id=UUID("24348246-6791-4ac0-9d86-b948cd6a0e46"),
+        host="cts.example.com",
+        raw_command=None,
+        smartapp_id=UUID("8dada2c8-67a6-4434-9dec-570d244e78ee"),
+        data={
```
Comment (lxmnk) [inline tests/system_events/test_smartapp.py:87]: Let's use simpler data, `{"foo": "bar"} for example.

Comment (Opti213) [reply tests/system_events/test_smartapp.py:87]: Maybe our protocol as example will be better?

**Review by Nimond — APPROVED** [review]

**Pull request merged**